### PR TITLE
Eth submitWork() rate is now hex string

### PR DESCRIPTION
**Pull request by LefterisJP**
Is based on https://github.com/ethereum/cpp-ethereum/pull/2847.

As per the [JSON RPC API doc](https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_submithashrate) submitHashrate should accept hexadecimal strings.

### CODE REVIEW

File: libweb3jsonrpc/WebThreeStubServerBase.cpp [hunk 1]
```diff
@@ -780,9 +780,9 @@ bool WebThreeStubServerBase::eth_submitWork(string const& _nonce, string const&,
 	}
 }
 
-bool WebThreeStubServerBase::eth_submitHashrate(int _hashes, string const& _id)
+bool WebThreeStubServerBase::eth_submitHashrate(std::string const& _hashes, string const& _id)
```
Comment (LianaHus) [inline libweb3jsonrpc/WebThreeStubServerBase.cpp]: you don't need std:: here

**Comment by akstunt** [comment]
Please lets us know when this all committed. BTW not just dev branches were effected.

File: libweb3jsonrpc/WebThreeStubServerBase.cpp [hunk 2]
```diff
@@ -780,9 +780,9 @@ bool WebThreeStubServerBase::eth_submitWork(string const& _nonce, string const&,
 	}
 }
 
-bool WebThreeStubServerBase::eth_submitHashrate(int _hashes, string const& _id)
+bool WebThreeStubServerBase::eth_submitHashrate(string const& _hashes, string const& _id)
 {
-	client()->submitExternalHashrate(_hashes, jsToFixed<32>(_id));
+	client()->submitExternalHashrate(stoi(_hashes, nullptr, 0), jsToFixed<32>(_id));
```
Comment (gavofyork) [inline libweb3jsonrpc/WebThreeStubServerBase.cpp]: should use `jsToInt`.

**Comment by LefterisJP** [comment]
Rebased on develop

**Comment by LefterisJP** [comment]
`jsToInt()` seems to be doing exactly what [std::stoi()](http://en.cppreference.com/w/cpp/string/basic_string/stol) with base of 0 does. Maybe it would be better to simplify it?

File: ethminer/MinerAux.h [hunk 3]
```diff
@@ -515,7 +515,7 @@ class MinerCLI
 					auto rate = mp.rate();
 					try
 					{
-						rpc.eth_submitHashrate((int)rate, "0x" + id.hex());
+						rpc.eth_submitHashrate(toHex((u256)rate, HexPrefix::Add), "0x" + id.hex());
```
Comment (gavofyork) [inline ethminer/MinerAux.h]: `toJS`?

**Pull request merged**